oxenstored/connection.ml

```diff
@@ -226,12 +208,24 @@ let get_path con =
   Printf.sprintf "/local/domain/%i/"
     (match con.dom with None -> 0 | Some d -> Domain.get_id d)
 
-let watch_create ~con ~path ~token =
+let get_watch_path_with_base base path =
+  if path.[0] = '@' || path.[0] = '/' then
+    path
+  else
+    base ^ path
```

Comment:
Can we be sure that there are no problems around a missing `/` or double `//`?

Reply:
this is an alternative version of `get_watch_path` below that avoids doing an unnecessary call, instead using a pre-calculated `base`. it's worked well so far since the validity of the path is also verified - double `//` are in particular rejected because splitting for `/` will then produce an empty string, which fails the `store.ml:name_is_valid`.